wandb/sdk/service/server.py

```diff
@@ -99,6 +99,8 @@ def _start_sock(self, mux: StreamMux) -> int:
         try:
             self._sock_server.start()
             port = self._sock_server.port
+            if self._pid:
+                mux.set_pid(self._pid)
```

Comment:
why do we need to reset the pid again?

Reply:
That is the only place we pass it into the StreamMux class for socket transport.  The other path was for grpc.

there is room for improvement to clean up this interface but i think it should wait until we have figured out how to support grpc and sock at the same time (if we want to)

Reply:
Yeah, makes sense. Thanks!